pkg/pbq/pbqmanager.go

```diff
@@ -0,0 +1,164 @@
+package pbq
+
+import (
+	"context"
+	"errors"
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/pbq/store"
+	"github.com/numaproj/numaflow/pkg/pbq/store/memory"
+	"github.com/numaproj/numaflow/pkg/pbq/store/noop"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sync"
+	"time"
+)
+
+var NonExistentPBQErr error = errors.New("missing PBQ for the partition")
+
+// Manager helps in managing the lifecycle of PBQ instances
+type Manager struct {
+	options *store.Options
+	pbqMap  map[string]*PBQ
+	log     *zap.SugaredLogger
+}
+
+// NewManager returns new instance of manager
+// We don't intend this to be called by multiple routines.
+func NewManager(ctx context.Context, opts ...store.SetOption) (*Manager, error) {
```

Comment:
we could make NewManager take PBQ options like `opts ...Options` and one of the option for the PBQ could be `WithStoreOptions(opts ...Store.Options)`.

WDYT?

Reply:
yeah it is better that way, I have added separate options for pbq now